### DISCUSSION TITLE
.APK decompiling

### DISCUSSION THREAD

**Discussion by middleboxs** [AssetRipper/AssetRipper#1521] [Q&A] [answered]
I'm trying to use it to decompile a VR game from quest 2, but since all quest 2 games are .APKs it isn't fully working. Is there a way to decompile .APKs or is it exclusively for .EXEs

**Comment by MiddleBOXESSS** [comment]
Different acc, I am not entirely sure if I even have the .APK, I have a folder structured: developername.gamename -> "cache" and "files" 
"cache" -> "vulkan_pso_cache.bin"  "files" -> "tombstone_000" "tombstone_01" "tombstone_02" (tombstone 1 and 2 are empty), "il2cpp"
"il2cpp" -> "unity.ver", "Metadata", "Resources"
"Metadata" -> "global-metadata.dat"
 "Resources" -> "mscorlib.dll-resources.dat"


I have looked throughout the entirety of the headset's files and found no other thing relating to the game, certainly no .APK, is there something I am missing?

**Reply by ds5678** [reply]
I've never taken an APK from a headset, but I'm sure there are guides online.

**Reply by MiddleBOXESSS** [reply]
Thanks, I managed to get the APK off the headset and decompile it, however the state it is in afterwards is not very useful, what are the most reliable settings for decompiling?

**Comment by ds5678** [comment]
AssetRipper is intended to work on any Unity game.

**Comment by middleboxs** [comment, marked as answer]
You can get .APKs off of quest 2 with sidequest app explorer, it can be decompiled from that if you don't include any other files I think, still working on properly decompiling the game to an explorable state.
